pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1705,25 +1687,11 @@ func (c *Cloud) GetVolumeLabels(volumeName KubernetesVolumeID) (map[string]strin
 	return labels, nil
 }
 
-// GetDiskPath implements Volumes.GetDiskPath
-func (c *Cloud) GetDiskPath(volumeName KubernetesVolumeID) (string, error) {
-	awsDisk, err := newAWSDisk(c, volumeName)
-	if err != nil {
-		return "", err
-	}
-	info, err := awsDisk.describeVolume()
-	if err != nil {
-		return "", err
-	}
-	if len(info.Attachments) == 0 {
-		return "", fmt.Errorf("No attachment to volume %s", volumeName)
-	}
-	return aws.StringValue(info.Attachments[0].Device), nil
-}
-
 // DiskIsAttached implements Volumes.DiskIsAttached
+// TODO: Remove this - we can inline it into DetachDisk (because it is only used there)
 func (c *Cloud) DiskIsAttached(diskName KubernetesVolumeID, nodeName types.NodeName) (bool, error) {
-	_, instance, err := c.getFullInstance(nodeName)
+	// NeverCache - but TODO remove this function
+	_, instance, err := c.getFullInstance(NeverCache, nodeName)
```

Comment:
I think one problem here is, this will pretty much always trigger a full instance cache refresh  - as it is currently implemented.

I think we should leave room for ability to read-through cache in certain cases and may be not force a full refresh because I am wondering if we would end up making more API requests than before.